core/src/main/java/nu/yona/server/messaging/service/DiscloseRequestMessageDTO.java

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.messaging.service;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import nu.yona.server.analysis.entities.GoalConflictMessage;
+import nu.yona.server.analysis.entities.GoalConflictMessage.Status;
+import nu.yona.server.messaging.entities.DiscloseRequestMessage;
+import nu.yona.server.messaging.entities.DiscloseResponseMessage;
+import nu.yona.server.messaging.entities.Message;
+import nu.yona.server.messaging.entities.MessageDestination;
+import nu.yona.server.messaging.service.MessageService.DTOManager;
+import nu.yona.server.messaging.service.MessageService.TheDTOManager;
+import nu.yona.server.subscriptions.service.UserDTO;
+
+@JsonRootName("discloseRequestMessage")
+public class DiscloseRequestMessageDTO extends MessageDTO
+{
+	private static final String ACCEPT = "accept";
+	private static final String REJECT = "reject";
+
+	private String nickname;
+	private boolean isAccepted;
+	private boolean isRejected;
+
```

Comment:
Can be done, I copied this from BuddyConnectRequestMessageDTO. If you like I will refactor both.